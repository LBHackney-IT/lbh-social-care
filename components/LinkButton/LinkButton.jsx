import PropTypes from 'prop-types';
import Router from 'next/router';

const LinkButton = ({ label, route }) => {
  const handleLink = (url) => window.open(url, '_blank');
  const isGoogleForm = route && route.includes('google');
  return (
    <button
      href="#"
      role="button"
      draggable="false"
      className={`govuk-button govuk-link-button govuk-!-margin-bottom-3`}
      data-module="govuk-button"
      onClick={() =>
        isGoogleForm ? handleLink(route) : Router.push(`${route}`)
      }
    >
      {label}
    </button>
  );
};

LinkButton.propTypes = {
  label: PropTypes.string.isRequired,
  route: PropTypes.string.isRequired,
};

export default LinkButton;
