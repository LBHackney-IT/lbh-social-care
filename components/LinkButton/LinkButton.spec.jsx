import { render } from '@testing-library/react';
import LinkButton from './LinkButton';

describe('LinkButton component', () => {
  const props = {
    label: 'Foo',
    route: 'https://foo/bar',
    internalQuery: '?foo',
  };
  it('should render properly', () => {
    const { asFragment } = render(<LinkButton {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
