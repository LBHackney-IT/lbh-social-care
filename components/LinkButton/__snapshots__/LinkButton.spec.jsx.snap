// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton component should render properly 1`] = `
<DocumentFragment>
  <button
    class="govuk-button govuk-link-button govuk-!-margin-bottom-3"
    data-module="govuk-button"
    draggable="false"
    href="#"
    role="button"
  >
    Foo
  </button>
</DocumentFragment>
`;
